### PR TITLE
Parallel GRU decoder with Banadau-style attention

### PR DESCRIPTION
**Pull request by tkornut**
- Fixes issues with description of hidden stream in decoder with attention (it actually required different order than defined in output_data_definitions)
- Standardized dimensions of input/output hidden steams in both RNN models: batch-major!

That, along with fixed sizes of embeddings (padding) enabled to properly parallelize both the model and C4 pipeline

### CODE REVIEW

**Comment by tkornut** [comment]
Early results:
 * 7 GPUs enabled me to use batch size of 300
 * on that setting, one epoch (i.e. processing of 4 folds of joined training and validation sets, ~3000 samples) takes around 30 seconds

Screen Shot 2019-05-03 at 4 31 06 PM
